File: client/java/src/avt/ImageObj.java
package avt;

import javax.microedition.lcdui.Graphics;
import javax.microedition.lcdui.Image;
import main.Canvas;

public final class ImageObj extends SubObject {
   private Image img;

   public ImageObj(int var1, int var2, int var3, int var4) {
      super(var1, var2, var3, 0);
      FilePack.b(T.at);
      this.img = FilePack.getImage("" + var1);
      if (this.img != null) {
         this.img.getWidth();
      }

      FilePack.reset();
   }

   public final void update() {
   }

   public final void paint(Graphics var1) {
      if (this.img == null) {
         super.g = AvatarData.getImgIcon((short)super.type).w;
         AvatarData.paintImg(var1, super.type, super.x * MyObject.hd, super.y * MyObject.hd, 33);
      } else {
         var1.drawImage(this.img, super.x * MyObject.hd, super.y * MyObject.hd, 33);
      }

      if (super.type == 846) {
         Canvas.fontChatB.drawString(var1, String.valueOf(MapScr.boardID), super.x * MyObject.hd, super.y * MyObject.hd - 30 * MyObject.hd, 2);
      } else if (super.type == 1029 && FarmScr.foodID != 0) {
         FarmItem var2 = FarmScr.getFarmItem(FarmData.getFoodByID(FarmScr.foodID).productID);
         String var3 = "";
         int var4;
         if ((var4 = FarmScr.remainTime / 3600) > 0) {
            var3 = var4 + ":";
         }

         int var5;
         if ((var5 = (FarmScr.remainTime - var4 * 3600) / 60) > 0 || var4 > 0) {
            var3 = var3 + var5 + ":";
         }

         var4 = FarmScr.remainTime - var4 * 3600 - var5 * 60;
         var3 = var3 + var4;
         if (FarmScr.remainTime == 0) {
            var3 = "hoan thanh";
         }

         FarmScr.xPosCook = super.x - Canvas.smallFontYellow.getWidth(var3) / 2 / MyObject.hd;
         FarmScr.yPosCook = super.y - AvatarData.getImgIcon((short)super.type).h / MyObject.hd - 10;
         FarmData.paintImg(var1, var2.IDImg, super.x * MyObject.hd - Canvas.smallFontYellow.getWidth(var3) / 2, super.y * MyObject.hd - AvatarData.getImgIcon((short)super.type).h - 10 * MyObject.hd, 3);
         Canvas.smallFontYellow.drawString(var1, var3, super.x * MyObject.hd - Canvas.smallFontYellow.getWidth(var3) / 2 + 10 * MyObject.hd, super.y * MyObject.hd - AvatarData.getImgIcon((short)super.type).h - 10 * MyObject.hd - AvMain.hSmall / 2 + 2, 0);
      }

   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/ImageObj.cs
public class ImageObj : SubObject
{
	public Image img;

	public ImageObj(int type, int x, int y, int w, int h)
		: base(type, x, y, w, h)
	{
		if (type >= 0)
		{
			img = Image.createImagePNG(T.getPath() + "/home/" + type);
		}
		if (img != null)
		{
			base.w = (short)img.w;
			base.h = (short)img.h;
		}
		else
		{
			base.w = (base.h = 0);
		}
	}

	public override void update()
	{
	}

	public override void paint(MyGraphics g)
	{
		if (img == null)
		{
			if (w == 0)
			{
				w = AvatarData.getImgIcon((short)type).w;
				h = AvatarData.getImgIcon((short)type).h;
			}
			AvatarData.paintImg(g, type, x * MyObject.hd, y * MyObject.hd, 33);
		}
		else
		{
			g.drawImage(img, x * MyObject.hd, y * MyObject.hd, 33);
		}
		if (type == 846)
		{
			Canvas.blackF.drawString(g, MapScr.boardID + string.Empty, x * MyObject.hd, y * MyObject.hd - 30 * MyObject.hd, 2);
		}
		else if (type == 1029 && FarmScr.foodID != 0)
		{
			Food foodByID = FarmData.getFoodByID(FarmScr.foodID);
			FarmItem farmItem = FarmScr.getFarmItem(foodByID.productID);
			string text = string.Empty;
			int num = FarmScr.remainTime / 3600;
			if (num > 0)
			{
				text = num + ":";
			}
			int num2 = (FarmScr.remainTime - num * 3600) / 60;
			if (num2 > 0 || num > 0)
			{
				text = text + num2 + ":";
			}
			int num3 = FarmScr.remainTime - num * 3600 - num2 * 60;
			text = text + num3 + string.Empty;
			if (FarmScr.remainTime == 0)
			{
				text = "Hoan thanh";
			}
			FarmScr.xPosCook = x - Canvas.smallFontYellow.getWidth(text) / 2 / MyObject.hd;
			int num4 = 60 * MyObject.hd;
			FarmScr.yPosCook = y - num4 / MyObject.hd - 10;
			FarmData.paintImg(g, farmItem.IDImg, x * MyObject.hd - Canvas.smallFontYellow.getWidth(text) / 2, y * MyObject.hd - num4 - 10 * MyObject.hd, 3);
			Canvas.smallFontYellow.drawString(g, text, x * MyObject.hd - Canvas.smallFontYellow.getWidth(text) / 2 + 10 * MyObject.hd, y * MyObject.hd - num4 - 10 * MyObject.hd - AvMain.hSmall / 2 + 2, 0);
		}
	}
}
